src/ValueSyntax.hs

```diff
@@ -0,0 +1,36 @@
+module ValueSyntax where
+
+import qualified Data.Map                      as M
+import qualified Data.Text                     as T
+
+import Syntax
+
+-- TODO: Think about Environment structure. How modules should be stored etc.
```

Comment:
This is indeed quite tricky and possible one of the hardest things to implement here. The environment should know about names defined in each module and about the relationships between them. We could assume that all imports are correct though.